### ISSUE TITLE
How can I return CSS with diff html while running in browser?

### ISSUE THREAD

**Issue by Coruscate5** [prettydiff/prettydiff#575]
I've been able to return HTML via **prettydiff.mode(options)**, using the 3rd party library setup (browser.js) inside my Angular 4 webapp, but it doesn't seem to include any CSS.

I've also tried setting "**complete_document**" to true, but that seems to throw an error regardless of which other options I set (it appears that "**finalFile**" doesn't exist, so it throws an error when trying to set the order).

Any ideas? I'm sure I'm just missing something simple

**Comment by prettydiff** [comment]
Pretty Diff is set to auto detect languages by default, but if it isn't happening for your CSS you could try:

```javascript
options.language = "css";
options.lexer = "style";
prettydiff.mode(options);
```

Would you be able to post a sample of the code as a gist so that I can reproduce the problem.  I am wondering why the language auto detection isn't working in this case.  I am also going to look into why it claims finalFile is missing.

**Comment by Coruscate5** [comment]
To clarify - I'm not trying to diff the CSS, I'm just diffing text/HTML, but I want to return CSS with the "diff_view" code so that it looks nice like it does on prettydiff.com (without having to copy-paste 1000 lines of CSS that is currently included at prettydiff.com).

I ended up just writing my own minimal CSS based on the diff html tag classes (e.g. "diff-left/diff-right").

Here's what my setup looks like: 

```
        var options =
            {
                api: "dom", //node, dom
                brace_line: false,
                brace_padding: false,
                brace_style: "none",
                braces: false,
                color: "white",
                comment_line: false,
                comments: false,
                complete_document: false,
                compressed_css: false,
                conditional: false,
                content: false,
                correct: false,
                crlf: false,
                css_insert_lines: false,
                diff: "",
                diff_comments: false,
                diff_context: -1,
                diff_format: "text", //text, html, json
                diff_label: "Edits",
                diff_rendered_html: false,
                diff_space_ignore: false,
                diff_view: "inline", //inline, sidebyside
                else_line: false,
                end: 0,
                end_comma: "never",
                end_quietly: "default",
                force_attribute: false,
                force_indent: false,
                format_array: "default",
                function_name: false,
                help: 80,
                indent_char: " ",
                indent_level: 0,
                indent_size: 4,
                jsscope: "none",
                language: "auto",
                language_default: "auto",
                language_name: "Markup",
                lexer: "auto", //markup, style, text, auto
                lexerOptions: {},
                list_options: false,
                method_chain: 3,
                minify_wrap: false,
                mode: "diff", //beautify, diff, minify, parse
                never_flatten: false,
                new_line: false,
                no_case_indent: false,
                no_lead_zero: false,
                node_error: false,
                object_sort: false,
                output: "",
                parse_format: "parallel",
                parse_space: false,
                preserve: 0,
                preserve_comment: false,
                preserve_text: false,
                quote: false,
                quote_convert: "none",
                read_method: "auto",
                selector_list: false,
                semicolon: false,
                source: "",
                source_label: "Original",
                space: true,
                space_close: false,
                start: 0,
                styleguide: "none",
                summary_only: false,
                tag_merge: false,
                tag_sort: false,
                ternary_line: false,
                top_comments: false,
                unformatted: false,
                variable_list: "none",
                version: false,
                vertical: false,
                wrap: 0
            };

        options.source = "SOME TEXT";
        options.diff = "MORE TEXT";

        var output = prettydiff.mode(options);
```

**Comment by prettydiff** [comment]
I see what you mean.  The `complete_document` option is what you want.  I will see tomorrow if I can produce any errors using that option.  Are you executing Pretty Diff from your browser or using Node.js?

**Comment by Coruscate5** [comment]
Diffing from browser - Chrome 72.0.3626.109

Just a side note - this engine seems awesome, but for us blue-collar code miners the documentation might be hard to follow. With my setup (Angular 4 with .NET Core backend, Webpack for bundling/compilation), it was surprisingly difficult for me to understand that all I need is one line:

`import 'prettydiff/js/browser.js';`

... and then futz around with the options in the API docs until I got something back that looked legible. Not really sure how to quantify this, but as an example of something that I didn't have any trouble with, check out TinyMCE's docs on including in webpack/usage/etc.

In case you're curious how I ended up with your lib, I got it from here: 
https://stackoverflow.com/questions/3053587/javascript-based-diff-utility

**Comment by prettydiff** [comment]
I have not tested the application in the use case of ES6 modules.  I am pleasantly surprised that worked.  I will review that and update the documentation.  I am investigating this right now.

**Comment by prettydiff** [comment]
After an investigation the functionality always seems to work for me.  I cannot reproduce the defect of the missing `finalFile` library.  I tested in both my localhost and from the domain.

If you are including the code into another application using `import` then I am guessing you are testing or executing the application on your localhost.  Everything should already be built if requested from NPM.  Trying running the build and see if that fixes the problem: `node js/services build`

I did however find some usability bugs with the output that will be updated later today in the next release.
